### PR TITLE
Update and add crate features

### PR DESCRIPTION
**Pull request by spring4175**
Update existing crate features and add details about new ones. Information is adapted from information in each crate's documentation.

### CODE REVIEW

**Comment by spring4175** [comment]
Blocks on #29

File: src/chapter_1_crates/section_2_http.md [hunk 1]
```diff
@@ -7,6 +7,45 @@ but it can be changed to use NativeTLS which uses the TLS native to the platform
...
 Ratelimiting is included out-of-the-box, along with support for proxies.
 
+## Features
+
+### Deserialization
+
+`twilight-gateway` supports [`serde_json`] and [`simd-json`] for deserializing
+and serializing events.
+
+#### Simd JSON
```
Comment (7596ff) [inline src/chapter_1_crates/section_2_http.md]: I think we should clarify the capitlization of SIMD here and the other mentions of it, I don't know what the right one is but the website uses all lowercase https://simdjson.org/

File: src/chapter_1_crates/section_7_first_party/section_4_util.md [hunk 2]
```diff
@@ -1,12 +1,47 @@
...
-ecosystem that do not fit in any other crate. Currently, it contains
-a trait to make extracting data from Discord identifiers (Snowflakes)
-easier.
+ecosystem that do not fit in any other crate. One example feature of the crate
+is a trait to make extracting data from Discord identifiers (snowflakes) easier.
+
+## Features
+
+`twilight-util` by default exports nothing. Features must be individually opted
+into via feature flags.
```
Comment (7596ff) [inline src/chapter_1_crates/section_7_first_party/section_4_util.md]:
```suggestion
`twilight-util` by default exports nothing. Features must be individually enabled
via feature flags.
```
(needs word wrapped)

File: src/chapter_1_crates/section_7_first_party/section_4_util.md [hunk 3]
```diff
@@ -1,12 +1,47 @@
...
+
+### Link
+
+Enables the parsing and formatting of URLs to resources, such as parsing and
+formatting webhook links or links to a user's avatar.
```
Comment (7596ff) [inline src/chapter_1_crates/section_7_first_party/section_4_util.md]: Should we name the flag again in `monospace`, like other flags?

**Review by 7596ff — CHANGES_REQUESTED** [review]

**Review by 7596ff — APPROVED** [review]

**Pull request merged**